app/auth/auth.py

```diff
@@ -59,10 +60,14 @@ async def get_authentication_context(
     context = None
     try:
         if actor_id.startswith(System.PK_PREFIX):
-            system = await system_handler.get(
-                actor_id,
-                [System.status == SystemStatus.ACTIVE],
-            )
+            try:
+                system = await system_handler.get(
+                    actor_id,
+                    [System.status == SystemStatus.ACTIVE],
+                )
+            except NotFoundError as exc:
```

Comment:
I don't think we need this try/except block since `NotFoundError` inherits from `DatabaseError` that is handled by outer try/except block. Removing it will reduce a bit the cognitive complexity.